Sources/StoreHelper/Core/PrePurchaseSubscriptionInfo.swift

```diff
@@ -22,30 +22,59 @@ public struct PurchasePriceForDisplay: Hashable, Identifiable {
 @available(iOS 15.0, macOS 12.0, *)
 public struct PrePurchaseSubscriptionInfo: Hashable {
     
-    /// The product's unique id.
-    public var productId: ProductId
-    
-    /// The product's display name.
-    public var name: String
-    
-    /// Localized standard (non-offer) price (e.g. "$1.99").
-    public var purchasePrice: String?
-    
-    /// The standard (non-offer) renewal period (e.g. "/ month").
-    public var renewalPeriod: String?
-    
+    /// The product for the subscription.
+    public let product: Product
+
+    /// The duration that this subscription lasts before auto-renewing.
+    public let subscriptionPeriod: Product.SubscriptionPeriod
+
     /// Info on the introductory offer, if any.
-    public var introductoryOffer: SubscriptionOfferInfo?
-    
+    public let introductoryOffer: SubscriptionOfferInfo?
+
     /// Info on any promotional offers. Should only be presented as an upgrade to users with a current subscription, or lapsed subscribers.
-    public var promotionalOffers: [SubscriptionOfferInfo]?
+    public let promotionalOffers: [SubscriptionOfferInfo]
 
     /// True if the `introductoryOffer` property may be presented to the user, false otherwise.
-    public var introductoryOfferEligible = false
-    
+    public internal(set) var introductoryOfferEligible = false
+
     /// True if the `promotionalOffers` property contains eligible offers for the user, false otherwise.
-    public var promotionalOffersEligible = false
-    
+    public internal(set) var promotionalOffersEligible = false
+
+    /// The unique product identifier.
+    public var productId: String {
+        product.id
+    }
+
+    /// A localized display name of the product.
+    public var name: String {
+        product.displayName
+    }
+
+    /// A localized string representation of `price`.
+    public var purchasePrice: String {
+        product.displayPrice
+    }
+
+    /// The price text for display (e.g. "$0.99 / 1 month", "$1.99 / 2 months").
+    public var displayPrice: String {
```

Comment:
Use Apple's recommended API for price display. The API can handle localization as well as singluar and plural units.

Reply:
Very good idea. This wasn't available when I first wrote the code so I had to create my display logic. Much better to use Apple's that supports proper localization.